File: benchmarks/method_dispatch/method_dispatch.cpp
#include <cstdio>

class Adder {
public:
  long long base;
  Adder(long long b) : base(b) {}
  long long add(long long n) { return base + n; }
};

int main() {
  Adder a(1);
  long long total = 0;
  long long i = 0;
  while (i < 10000000LL) {
    total += a.add(i);
    i += 1;
  }
  std::printf("%lld\n", total);
  return 0;
}

File: benchmarks/method_dispatch/method_dispatch.rs
struct Adder {
  base: i64,
}

impl Adder {
  fn add(&self, n: i64) -> i64 {
    self.base + n
  }
}

fn main() {
  let a = Adder { base: 1 };
  let mut total: i64 = 0;
  let mut i: i64 = 0;
  while i < 10_000_000 {
    total += a.add(i);
    i += 1;
  }
  println!("{total}");
}
